seastar-macros/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+#[proc_macro_attribute]
+pub fn test(
+    args: proc_macro::TokenStream,
+    item: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+    let input = syn::parse_macro_input!(item as syn::ItemFn);
+
+    let name = &input.sig.ident;
+    let body = &input.block;
+    let attrs = &input.attrs;
+
+    if input.sig.asyncness.is_none() {
+        let msg = "the async keyword is missing from the function declaration";
+        return syn::Error::new_spanned(input.sig.fn_token, msg)
+            .to_compile_error()
+            .into();
+    }
+
+    if !args.is_empty() {
+        let msg = "arguments for #[seastar::test] are not supported yet";
+        return syn::Error::new_spanned(input.sig.inputs, msg)
```

Comment:
I've never written a proc macro so take those comments with a grain of salt.
Why use `input.sig.inputs` instead of `args`? The formar also has `ToTokens` implementation, so you should be able to use it, and `args` is the actual place where the error is, not function signature.